test_auc_gold_labels_behaviour in auc_test.py fails intermittently

**Describe the bug**
The `test_auc_gold_labels_behaviour` test in  `allennlp/tests/training/metrics/auc_test.py::AucTest` fails intermittently when the generated labels contains all `3s`.

Error message:

```python
==================================================================================================== FAILURES ====================================================================================================
_____________________________________________________________________________________ AucTest.test_auc_gold_labels_behaviour _____________________________________________________________________________________

self = <allennlp.tests.training.metrics.auc_test.AucTest testMethod=test_auc_gold_labels_behaviour>

   
    def test_auc_gold_labels_behaviour(self):
        auc = Auc(positive_label=4)
        predictions = torch.randn(8).float()
        labels = torch.randint(3, 5, (8,)).long()
>       auc(predictions, labels)

allennlp/tests/training/metrics/auc_test.py:50: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <allennlp.training.metrics.auc.Auc object at 0x7f548d86f438>, predictions = tensor([ 0.4905, -2.1212,  1.6025, -1.8595, -1.5871,  1.8005,  0.0391, -0.0910])
gold_labels = tensor([3, 3, 3, 3, 3, 3, 3, 3]), mask = None

    def __call__(
        self,
        predictions: torch.Tensor,
        gold_labels: torch.Tensor,
        mask: Optional[torch.Tensor] = None,
    ):
        """
        Parameters
        ----------
        predictions : ``torch.Tensor``, required.
            A one-dimensional tensor of prediction scores of shape (batch_size).
        gold_labels : ``torch.Tensor``, required.
            A one-dimensional label tensor of shape (batch_size), with {1, 0}
            entries for positive and negative class. If it's not binary,
            `positive_label` should be passed in the initialization.
        mask: ``torch.Tensor``, optional (default = None).
            A one-dimensional label tensor of shape (batch_size).
        """
    
        predictions, gold_labels, mask = self.unwrap_to_tensors(predictions, gold_labels, mask)
    
        # Sanity checks.
        if gold_labels.dim() != 1:
            raise ConfigurationError(
                "gold_labels must be one-dimensional, "
                "but found tensor of shape: {}".format(gold_labels.size())
            )
        if predictions.dim() != 1:
            raise ConfigurationError(
                "predictions must be one-dimensional, "
                "but found tensor of shape: {}".format(predictions.size())
            )
    
        unique_gold_labels = torch.unique(gold_labels)
        if unique_gold_labels.numel() > 2:
            raise ConfigurationError(
                "AUC can be used for binary tasks only. gold_labels has {} unique labels, "
                "expected at maximum 2.".format(unique_gold_labels.numel())
            )
    
        gold_labels_is_binary = set(unique_gold_labels.tolist()) <= {0, 1}
        if not gold_labels_is_binary and self._positive_label not in unique_gold_labels:
            raise ConfigurationError(
                "gold_labels should be binary with 0 and 1 or initialized positive_label "
>               "{} should be present in gold_labels".format(self._positive_label)
            )
E           allennlp.common.checks.ConfigurationError: 'gold_labels should be binary with 0 and 1 or initialized positive_label 4 should be present in gold_labels'

allennlp/training/metrics/auc.py:68: ConfigurationError
```

**To Reproduce**
This can be reproduced by setting the following seed in the test:
```python
   def test_auc_gold_labels_behaviour(self):        
        torch.manual_seed(1178076155441788456)
        auc = Auc(positive_label=4)
        predictions = torch.randn(8).float()
        labels = torch.randint(3, 5, (8,)).long()
        auc(predictions, labels)
```

**Expected behavior**
The test should always pass

**System (please complete the following information):**
```
OS: ubuntu 18.04
torch==1.3.1
numpy==1.18.0
pytest==5.3.2
python 3.6.9

allennlp: installed from source; commit 95ef61

```

**Additional context**
The test fails at least 2 times out of 30 times I tried, each time using a different seed

--- a/allennlp/tests/training/metrics/auc_test.py
+++ b/allennlp/tests/training/metrics/auc_test.py
@@ -48,9 +48,9 @@ class AucTest(AllenNlpTestCase):
         auc = Auc(positive_label=4)
 
         predictions = torch.randn(8).float()
-        labels = torch.randint(3, 5, (8,)).long()
+        labels = torch.randint(0, 5, (8,)).long()
         # We make sure that the positive label is always present.
-        labels[torch.randint(0, 8, (1,))] = 4
+        labels[0] = 4
         auc(predictions, labels)
         computed_auc_value = auc.get_metric(reset=True)
 

--- a/allennlp/tests/training/metrics/auc_test.py
+++ b/allennlp/tests/training/metrics/auc_test.py
@@ -50,6 +50,8 @@ class AucTest(AllenNlpTestCase):
         predictions = torch.randn(8).float()
         labels = torch.randint(3, 5, (8,)).long()
 
+        if 4 not in labels:
+            labels[torch.randint(0, 8, (1,))]=4
         auc(predictions, labels)
         computed_auc_value = auc.get_metric(reset=True)
 

--- a/allennlp/tests/training/metrics/auc_test.py
+++ b/allennlp/tests/training/metrics/auc_test.py
@@ -49,9 +49,8 @@ class AucTest(AllenNlpTestCase):
 
         predictions = torch.randn(8).float()
         labels = torch.randint(3, 5, (8,)).long()
-
-        if 4 not in labels:
-            labels[torch.randint(0, 8, (1,))]=4
+        # We make sure that the positive label is always present.
+        labels[torch.randint(0, 8, (1,))] = 4
         auc(predictions, labels)
         computed_auc_value = auc.get_metric(reset=True)
 

--- a/allennlp/tests/training/metrics/auc_test.py
+++ b/allennlp/tests/training/metrics/auc_test.py
@@ -48,7 +48,7 @@ class AucTest(AllenNlpTestCase):
         auc = Auc(positive_label=4)
 
         predictions = torch.randn(8).float()
-        labels = torch.randint(0, 5, (8,)).long()
+        labels = torch.randint(3, 5, (8,)).long()
         # We make sure that the positive label is always present.
         labels[0] = 4
         auc(predictions, labels)

